picobot/video_editor.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+import ffmpeg
+
+
+def sticker_from_video(mp4_path: Path):
+    '''
+    Converts the given video to a proper format that can be uploaded as a sticker.
+    Telegram accepts WEBM VP9 with a maximum size of 512x512 pixels with maximum 3 seconds duration.
+    '''
+    probe = ffmpeg.probe(mp4_path)
+    video_stream = next(
+        (stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None
+    )
+    width = int(video_stream['width'])
+    height = int(video_stream['height'])
+    duration = float(probe['format']['duration'])
+
+    check_duration(duration)
```

Comment:
Hmmm I'm not sure I would split this into a function as it is literally just a simple straight-forward `if`. Personally I would bring the function's code here:

```suggestion
    if duration > 3.0:
            raise ValueError('Video duration exceeds limits')
```